R/allelic_series_sumstats.R

```diff
@@ -0,0 +1,331 @@
+# Purpose: Implement sumstats-based allelic series test.
+# Updated: 2024-07-31
+
+# Default weights.
+DEFAULT_WEIGHTS <- c(1, 2, 3)
+
+#' Allelic Series Burden Test from Summary Statistics
+#' 
+#' Allelic series burden test from summary statistics. 
+#'
+#' @param anno (snps x 1) annotation vector with values in c(0, 1, 2).
+#' @param beta (snps x 1) vector of effect sizes for the coding genetic variants
+#'   within a gene.
+#' @param se (snps x 1) vector of standard errors for the effect sizes.
+#' @param check Run input checks? Default: TRUE.
+#' @param ld (snps x snps) matrix of correlations among the genetic variants.
+#'   Although ideally provided, an identity matrix is assumed if not.
+#' @param maf (snps x 1) vector of minor allele frequencies. Although ideally
+#'   provided, defaults to the zero vector.
+#' @param method Method for aggregating across categories:
+#'   {"none", "sum"}. Default: "none".
+#' @param weights (3 x 1) vector of annotation category weights.
+#' @return Numeric p-value of the allelic series burden test.
+#' @examples
+#' # Generate data.
+#' data <- DGP(n = 1e3)
+#' sumstats <- CalcSumstats(data = data)
+#' 
+#' # Run allelic series burden test from sumstats.
+#' results <- ASBTSS(
+#'   anno = sumstats$anno,
+#'   beta = sumstats$sumstats$beta, 
+#'   maf = sumstats$maf,
+#'   se = sumstats$sumstats$se,
+#'   ld = sumstats$ld
+#' )
+#' show(results)
+#' @export
+ASBTSS <- function(
+  anno,
+  beta,
+  se,
+  check = TRUE,
+  ld = NULL,
```

Comment:
In this function, when LD = identity is used, do we want to add the functionality to apply genomic control with some lambda?

Reply:
Wanted to ask how you thought we should implement this: we cannot determine the inflation factor based on the results of a single test, but I think we could add the option to supply an inflation factor which has been determined previously. Is that what you had in mind?

Reply:
Yes, this is what I meant.

Reply:
Added an option for genomic control.